subcommand/inject-connect/command.go

```diff
@@ -323,15 +343,73 @@ func (c *Command) Run(args []string) int {
 		TLSConfig: &tls.Config{GetCertificate: c.getCertificate},
 	}
 
-	c.UI.Info(fmt.Sprintf("Listening on %q...", c.flagListen))
-	if err := server.ListenAndServeTLS("", ""); err != nil {
-		c.UI.Error(fmt.Sprintf("Error listening: %s", err))
-		return 1
+	// channel used for health checks
+	// also check to see if we should enable TLS
+	consulAddr := os.Getenv("CONSUL_HTTP_ADDR")
```

Comment:
I don't think we can assume that the consul address will always come from the env variable. You can also pass it via the `-http-addr` http flag:
```  
-http-addr=<address>
     The `address` and port of the Consul HTTP agent. The value can be
     an IP address or DNS address, but it must also include the port.
     This can also be specified via the CONSUL_HTTP_ADDR environment
     variable. The default value is http://127.0.0.1:8500. The scheme
     can also be set to HTTPS by setting the environment variable
     CONSUL_HTTP_SSL=true.
```

Reply:
This will always be the case when healthChecks are enabled, corresponding code to pass it is in helm.

Reply:
Hmm, what if I pass just the command I mentioned? I know we're setting the env variable in the helm chart, but what I'm saying is that I don't think we can assume this will always be the case when this command is run.

Reply:
Isn't that a bit of a stretch?
If the command is run and healthchecks have been enabled through helm it will have been passed, any other way of enabling this code path implies the user is bypassing helm. Do we have the protection for every combination of inputs on all subcommands? Are we trying to support deploying without helm?